### PR TITLE
Story1 collection

### PR DESCRIPTION
**Pull request by BuzatuCos**
👍

### CODE REVIEW

File: Test/functions.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        $expected = '<div class="rows"><p class="brand">golf</p><p class="model">vw : jetta</p><p class="year">volk : 3213</p></div><div class="rows"><p class="brand">passat</p><p class="model">opel : kadett</p><p class="year">volk : 3893</p></div>';
+        $input =[['wv'=>'golf','vw'=>'jetta','volk'=>'3213'],['wv'=>'passat','opel'=>'kadett','volk'=>'3893']];
+        $case = extractDataForOutput($input);
+        $this->assertEquals($expected, $case);
+    }
+
+    public function testFailureextractDataForOutput()
+    {
+        $expected ='<div class="rows"><p class="brand">321</p><p class="model">22 : jetta</p><p class="year">volk : 3213</p></div><div class="rows"><p class="brand">passat</p><p class="model">55 : kadett</p><p class="year">volk : 3893</p></div>';
+        $input =[['321'=>'321','22'=>'jetta','volk'=>'3213'],['wv'=>'passat','55'=>'kadett','volk'=>'3893']];
```
Comment (charliecog) [inline Test/functions.php]: this isnt a failure test, try giving it an array with a string in it for example: ['this', 'that'] and see how it handles it

File: db/cosminCollection.sql [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+
+# Dump of table Cars
+# ------------------------------------------------------------
+
+DROP TABLE IF EXISTS `Cars`;
+
+CREATE TABLE `Cars` (
+  `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
+  `Brand` varchar(20) DEFAULT NULL,
```
Comment (charliecog) [inline db/cosminCollection.sql]: make the length longer eg: 255

File: db/cosminCollection.sql [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  `Model` varchar(20) DEFAULT NULL,
```
Comment (charliecog) [inline db/cosminCollection.sql]: make the length longer eg: 255

File: functions.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
+<?php
+/**create a function to connect to the database
+ * add attribute that will allow to extract only associative arrays
+ *
+ * @return PDO the database connection
+ */
+function dbConn():PDO
+{
```
Comment (charliecog) [inline functions.php]: should be on the line above

File: index.php [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+?>
+<!DOCTYPE html>
+<html lang="en-gb">
+<head>
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" type="text/css" href="styles.css" />
+    <title>Collection</title>
+</head>
+<body>
+<h1>Car Collection</h1>
```
Comment (charliecog) [inline index.php]: indentation

Comment (charliecog) [reply index.php]: for the whole file

File: styles.css [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+.section {
+	width:1200px;
+	overflow:auto;
+	margin:0 auto;
+}
+
+
+.rows {
+	overflow:auto;
```
Comment (charliecog) [inline styles.css]: not doing anything

File: styles.css [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+	width:30%;
+	margin: 0 auto;
```
Comment (charliecog) [inline styles.css]: not doing anything

File: styles.css [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+	text-align: center;
+	float:left;
```
Comment (charliecog) [inline styles.css]: need a space between float: and left;

**Review by charliecog — COMMENTED** [review]
general comment: don't allow NULL in the db for your data as it would allow empty fields to be inserted by the user

also you will need to include the normalize.css file as a best practice

File: styles.css (hunk 8 above)
Comment (BuzatuCos) [reply styles.css]: fix

File: styles.css (hunk 7 above)
Comment (BuzatuCos) [reply styles.css]: fix

File: styles.css (hunk 6 above)
Comment (BuzatuCos) [reply styles.css]: fix

File: index.php (hunk 5 above)
Comment (BuzatuCos) [reply index.php]: fix

File: functions.php (hunk 4 above)
Comment (BuzatuCos) [reply functions.php]: fix

File: db/cosminCollection.sql (hunk 3 above)
Comment (BuzatuCos) [reply db/cosminCollection.sql]: fix

File: db/cosminCollection.sql (hunk 2 above)
Comment (BuzatuCos) [reply db/cosminCollection.sql]: fix

File: functions.php [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+ */
+function extractDataForOutput(array $result): string
+{
+    $output = '';
+    foreach ($result as $array) {
+        $key = array_keys($array);
+        if($key[0] !== 'Brand'){
+            return 'Error';
+        }
+        $output .= '<div class="rows">' . '<p class="brand">' . $array[$key[0]] . '</p>' . '<p class="model">' . $key[1] . ' : ' . $array[$key[1]] . '</p>' . '<p class="year">' . $key[2] . ' : ' . $array[$key[2]] . '</p>' . '</div>';
```
Comment (charliecog) [inline functions.php]: just hard code the keys dont need to get them from db

File: functions.php [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+ *
+ * @param array $result a multidimensional array generated from the database
+ *
+ * @return string extracting data from the array preparing for the html including div and p with different classes
+ */
+function extractDataForOutput(array $result): string
+{
+    $output = '';
+    foreach ($result as $array) {
+        $key = array_keys($array);
```
Comment (charliecog) [inline functions.php]: just hard code the keys dont need to get them from db

**Review by charliecog — COMMENTED** [review]
almost there!

File: functions.php [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+ *
+ * @param array $result a multidimensional array generated from the database
+ *
+ * @return string extracting data from the array preparing for the html including div and p with different classes
+ */
+function extractDataForOutput(array $result): string
+{
+    $output = '';
+    foreach ($result as $array) {
+        if(array_key_exists('Brand',$array) === false && array_key_exists('Model',$array) === false && array_key_exists('Year',$array) === false) {
```
Comment (charliecog) [inline functions.php]: use OR instead of AND

**Review by charliecog — APPROVED** [review]
Nice work!

**Pull request merged**